src/box/sql/mem.c

```diff
@@ -1854,11 +1854,11 @@ mem_get_dec(const struct Mem *mem, decimal_t *d)
 		return 0;
 	}
 	if (mem->type == MEM_TYPE_INT) {
-		decimal_from_int64(d, mem->u.r);
+		decimal_from_int64(d, mem->u.i);
 		return 0;
 	}
 	if (mem->type == MEM_TYPE_UINT) {
-		decimal_from_int64(d, mem->u.u);
+		decimal_from_uint64(d, mem->u.u);
```

Comment:
This doesn't look like a security issue. It's just a bug. I think we should move the ticket to the open-source repository.

Reply:
Filled an issue.